### PR TITLE
Minor: Add some examples to Date*Array and Time*Array

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
N/a 

# Rationale for this change
 
I was a little confused while reviewing https://github.com/apache/arrow-rs/pull/3647 so I wanted to clarify

# What changes are included in this PR?
Add some examples to the docs

# Are there any user-facing changes?

docs

### CODE REVIEW

File: arrow-array/src/array/primitive_array.rs [hunk 1]
```diff
@@ -168,22 +168,42 @@ pub type TimestampNanosecondArray = PrimitiveArray<TimestampNanosecondType>;
...
+/// representing the elapsed time since UNIX epoch in days."
+///
+/// This type is similar to the [`chrono::NaiveDate`] type and can hold
+/// values such as `2018-11-13`
 pub type Date32Array = PrimitiveArray<Date32Type>;
-/// A primitive array where each element is of 64-bit date type.
+/// A primitive array where each element is a 64-bit value
+/// representing the elapsed time since the UNIX epoch in milliseconds.
+///
+/// This type is similar to the [`chrono::DateTime`] type and can hold
```
Comment (tustvold) [inline arrow-array/src/array/primitive_array.rs:180]: NaiveDateTime perhaps?

**Review by tustvold — APPROVED** [review]

File: arrow-array/src/array/primitive_array.rs [hunk 2]
```diff
@@ -168,22 +168,42 @@ pub type TimestampNanosecondArray = PrimitiveArray<TimestampNanosecondType>;
...
 /// since midnight.
+///
+/// This type is similar to the [`chrono::NaiveTime`] type and can
+/// hold values such as `00:02:00.123`
 pub type Time32MillisecondArray = PrimitiveArray<Time32MillisecondType>;
 /// An array where each element is of 64-bit type representing time elapsed in microseconds
 /// since midnight.
+///
+/// This type is similar to the [`chrono::NaiveTime`] type and can
+/// hold values such as `00:02:00.000123`
```
Comment (viirya) [inline arrow-array/src/array/primitive_array.rs:200]:
```suggestion
/// hold values such as `00:02:00.123456`
```

File: arrow-array/src/array/primitive_array.rs [hunk 3]
```diff
@@ -168,22 +168,42 @@ pub type TimestampNanosecondArray = PrimitiveArray<TimestampNanosecondType>;
...
 pub type Time64MicrosecondArray = PrimitiveArray<Time64MicrosecondType>;
 /// An array where each element is of 64-bit type representing time elapsed in nanoseconds
 /// since midnight.
+///
+/// This type is similar to the [`chrono::NaiveTime`] type and can
+/// hold values such as `00:02:00.000000123`
```
Comment (viirya) [inline arrow-array/src/array/primitive_array.rs:206]:
```suggestion
/// hold values such as `00:02:00.123456789`
```

Comment (viirya) [reply arrow-array/src/array/primitive_array.rs:206]: easier to read. 😄

**Review by viirya — APPROVED** [review]

File: arrow-array/src/array/primitive_array.rs (hunk 1 above)
Comment (alamb) [reply arrow-array/src/array/primitive_array.rs]: in a685bf8d3

**Pull request merged**